Remove exit calls from ot_sender and ot_receiver

markspanbroek opened the pull request:
Use bool return value to indicate success or failure.

Prevents crashing an application that uses SimpleOT when wrong input is received.

mkskeller commented:
Hi, I don't want to merge this is in the current form because it changes the "contract", that is, it requires to change existing applications to handle the error themselves. I would suggest that you create a new function with the new behaviour, which is in turn used by the existing functions.

markspanbroek committed:
Introduce receiver_procS_check() and sender_keygen_check()

These functions allow a caller to check whether the input was correct
through the boolean return value. By contrast, the functions
receiver_procS() and sender_keygen() crash the application when input is
incorrect.

markspanbroek commented:
That's a good idea. I pushed some changes to address this.
